src/server/lib/middleware/login.ts

```diff
@@ -10,9 +10,78 @@ import { logger } from '@/server/lib/serverSideLogger';
 import { addQueryParamsToUntypedPath } from '@/shared/lib/queryParams';
 import { ResponseWithRequestState } from '@/server/models/Express';
 import { buildUrl } from '@/shared/lib/routeUtils';
+import {
+  performAuthorizationCodeFlow,
+  scopesForApplication,
+} from '@/server/lib/okta/oauth';
+import { RoutePaths } from '@/shared/model/Routes';
+import { ProfileOpenIdClientRedirectUris } from '@/server/lib/okta/openid-connect';
+import {
+  deleteOAuthTokenCookie,
+  getOAuthTokenCookie,
+  verifyAccessToken,
+  verifyIdToken,
+} from '@/server/lib/okta/tokens';
 
 const profileUrl = getProfileUrl();
 
+export const loginMiddlewareOAuth = async (
+  req: Request,
+  res: ResponseWithRequestState,
+  next: NextFunction,
+) => {
+  if (res.locals.queryParams.useIdapi) {
+    return loginMiddleware(req, res, next);
+  }
+
+  // if a user has the GU_SO cookie, they have recently signed out
+  // so we need to clear any existing tokens
+  // and perform the auth code flow to get new tokens to
+  // get new tokens and log the user in if necessary
```

Comment:
May have been fixed in a later commit: should be "and perform the auth code flow to get new tokens and log the user in if necessary"

Reply:
Updated the comment